docs/UserGuide.md

```diff
@@ -1,156 +1,224 @@
 
 > **Command Format**
->
-> * Words in `UPPER_CASE` are the parameters.
-> * Items in `SQUARE_BRACKETS` are optional.
-> * Items with `...` after them can have multiple instances.
-> * Parameters can be in any order.
-
-### 2.1. Viewing help : `help`
+> * All commands begin with a command word, following by parameters.
+> * All parameters are key-value pairs except the first parameter, whose key must be omitted.
+> * All keys start with the backslash `\`.
+> e.g. `\from`, `\every`.
+> * If you want to use backslash in the value, use `\\`.
+> * Both command word and keys are strings without space. Underscore is used separete different words.
+> e.g. `list_tag`, `\remind_at`
+> * Words in `UPPER_CASE` are the values.
+> * (A|B) means A and B can be used interchangeably.
+> * Items in `[]` are optional.
+> * Items with `...` after them can have multiple instances, separated by space.
+> * Key-value pairs can be in any order.
+
+### Viewing help : `help`
 
 Format: `help`
 
-> Help is also shown if you enter an incorrect command e.g. `abcd`
+> Help is also shown if the command format is invalid. e.g. `abcd`
 
-### 2.2. Adding a person: `add`
+### Adding a new task : `add`, `do`
 
-Adds a person to the address book<br>
-Format: `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...`
+Adds a new task into the to-do list<br>
+Format 1: `(add|do) TASK_DESCRIPTION [\from START_TIME] [\to END_TIME] [\remind_at REMIND_TIME] [\every RECURRENCE_INTERVAL] [\as PRIORITY] [\under TAG...]`
+Format 2: `(add|do) TASK_DESCRIPTION [\by TIME] [\remind_at REMIND_TIME] [\every RECURRENCE_INTERVAL] [\as PRIORITY] [\under TAG...]`
 
-> Persons can have any number of tags (including 0)
+> - Tasks can have any number of tags (including 0)
+> - The start time and remind time must be earlier or equal to the end time
+> - Tasks can have neither start time and end time. They are floating tasks
+> - Tasks with the same start time and end time or are only associated with one timing are deadlines
+> - For format 2, start time and end time will both be set to `TIME`
+> That is, `add ... by 8pm` is equivalent to `add ... from 8pm to 8pm`
+> - Tasks can have different start time and end time. They are events
+> - `Priority` can be `normal`, `important`, `very important`
+> - If you do not specify the `priority` of the task, it will be set to a default value of `normal`!
+> - If you specify a recurrence interval, every time a task ends or is finished, the start time, end time and remind time of the task will be updated to its next occurrence.
 
 Examples:
 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
-* `add Betsy Crowe t/friend e/betsycrowe@gmail.com a/Newgate Prison p/1234567 t/criminal`
-
-### 2.3. Listing all persons : `list`
+* `do group meeting \from 4pm today \to 6pm today \remind_at 3pm \as important \under school_work CS2103T`
+* `add submit pre-tutorial activities \by 6pm this Wednesday \remind_at 5pm this Wednesday \every week`
```

Comment:
- How about covering the basic date format for example instead of natural language kind of date (today, tomorrow, wednesday) first?
- This is a little complicated here. Try not to have multiple command names first. Or allow aliases for command instead.